research/conditional/utils/model_utils.py

```diff
@@ -513,15 +514,75 @@ def get_ff_layer(args):
     return return_fn
 
 
+def get_vanilla_mamba_layer(args):
+    import mamba_ssm
+
+    return lambda: mamba_ssm.Mamba(
+        d_model=args.dmodel, expand=args.mamba_expansion, use_fast_path=False
+    )
+
+
 def get_mamba_layer(args):
     import mamba_ssm
 
     if args.mamba_mode == "vanilla":
         return_fn = lambda: mamba_ssm.Mamba(
-            d_model=args.dmodel, expand=args.mamba_expansion
+            d_model=args.dmodel, expand=args.mamba_expansion, use_fast_path=False
```

Comment:
Somewhat changes previous behaviour of vanilla mamba

Reply:
valid point but I think it's not significant, almost same steps/s afaik. I wanted to keep `use_fast_path=False` for consistency and modified Mamba requires it